rust/processor/src/processors/econia_processor.rs

```diff
@@ -0,0 +1,200 @@
+use super::processor_trait::{ProcessingResult, ProcessorTrait};
+use crate::{
+    models::default_models::transactions::{TransactionDetail, TransactionModel},
+    utils::database::{execute_with_better_error, PgDbPool},
+};
+
+use aptos_indexer_protos::transaction::v1::Transaction;
+use async_trait::async_trait;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Utc};
+use diesel::{result::Error, PgConnection};
+use econia_db::models::MarketRegistrationEvent;
+use econia_db::schema::market_registration_events;
+use serde_json::Value;
+use std::str::FromStr;
+use std::{collections::HashMap, fmt::Debug};
+
+pub const NAME: &str = "econia_processor";
+pub struct EconiaTransactionProcessor {
+    connection_pool: PgDbPool,
+    econia_address: String,
+}
+
+impl EconiaTransactionProcessor {
+    pub fn new(connection_pool: PgDbPool, econia_address: String) -> Self {
+        Self {
+            connection_pool,
+            econia_address,
+        }
+    }
+}
+
+impl Debug for EconiaTransactionProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = &self.connection_pool.state();
+        write!(
+            f,
+            "DefaultTransactionProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+fn hex_to_string(hex: &str) -> String {
+    let hex_no_prefix = &hex[2..];
+    let hex_bytes = hex::decode(hex_no_prefix).unwrap();
+    String::from_utf8(hex_bytes).unwrap()
+}
+
+fn opt_value_to_big_decimal(value: Option<&Value>) -> BigDecimal {
+    let lot_size_str = value
+        .and_then(Value::as_str)
+        .expect("key not found or not a string");
+    BigDecimal::from_str(lot_size_str).expect("Failed to parse BigDecimal")
+}
+
+fn value_is_not_empty_string(val: &Value) -> bool {
+    val.as_str().expect("Value is not a string").is_empty() == false
+}
+
+fn insert_market_registration_events(
+    conn: &mut PgConnection,
+    events: Vec<MarketRegistrationEvent>,
+) -> Result<(), diesel::result::Error> {
+    execute_with_better_error(
+        conn,
+        diesel::insert_into(market_registration_events::table)
+            .values(&events)
+            .on_conflict_do_nothing(),
```

Comment:
What are the implications of doing nothing?

If we try to insert but it doesn't work will we miss an event?

Reply:
If we try to insert an event twice, as according to its transaction version and event index, the second insertion will just be dropped and lost to the wind. Added this as a comment.